decode.go

```diff
@@ -18,30 +18,58 @@ type Unmarshaler interface {
 
 // Unmarshal parses the plist-encoded data and stores the result in the value pointed to by v.
 func Unmarshal(data []byte, v interface{}) error {
-	return NewDecoder(bytes.NewReader(data)).Decode(v)
+	// Check for binary plist here before setting up the decoder.
+	if string(data[:7]) == "bplist0" {
+		return NewBinaryDecoder(bytes.NewReader(data)).Decode(v)
+	}
+	return NewXMLDecoder(bytes.NewReader(data)).Decode(v)
 }
 
 // A Decoder reads and decodes Apple plist objects from an input stream.
+// The plists can be in XML or binary format.
 type Decoder struct {
-	reader io.Reader
+	reader   interface{} // io.Reader for XML, io.ReadSeeker for binary
```

Comment:
We can set the field to a ReadSeeker and default to a `noop` implementation of Seek if the plist is in XML format. 
Thoughts? I'd like to avoid empty interface where possible.

Reply:
My go-fu is not great and I was struggling with how to do this the right way.  I didn't want to restrict the xml decoder to being an io.ReadSeeker and adding a noop Seek seems kind of shady.  I've changed reader back to an io.Reader and then just later assert it up to an io.ReadSeeker when needed for the binary decoder, which is maybe better?